spek-ide-plugin/intellij-common/src/main/kotlin/org/spekframework/intellij/SpekRunConfigurationProducer.kt

```diff
@@ -31,14 +35,44 @@ abstract class SpekRunConfigurationProducer(val producerType: ProducerType, conf
 ) {
     override fun isConfigurationFromContext(configuration: SpekBaseRunConfiguration<*>,
                                             context: ConfigurationContext): Boolean {
-        val path = context.psiLocation?.let { extractPath(it, true) }
-        return configuration.path == path
+        val descriptor = context.psiLocation?.let {
+            val elementContext = PsiTreeUtil.getContextOfType(it, KtClassOrObject::class.java, KtCallExpression::class.java)
+            if (elementContext != null) {
```

Comment:
If you're doing the `is` checks below you don't need to check for `!= null` and hence reduce the number of depth